src/team/dynamical_movement_primitives.py

```diff
@@ -125,27 +125,23 @@ def __init__(
     @classmethod
     def load_dmp(cls, dir_path: Path) -> "DynamicMovementPrimitives":
         """
-        Creates an DynamicMovementPrimitives with parameters loaded from a dmp param
-        file and regression information loaded from a regression file.
+        Creates an DynamicMovementPrimitives with dmp parameters and regression loaded from a dmp data
```

Comment:
This is nice! I want to propose a little improvement thought because I was also working on it ;): I thnk we should keep the old method and create a new save that will translate the old format if needed:

```python
   @classmethod
    def load(cls, dir_path: Path) -> "DynamicMovementPrimitives":
        """Load data saved with `self.save_dmp()`

        :param dir_path: folder where the data is saved
        :return: DynamicMovementPrimitives with the loaded parameters
        """
        # Check if old format data
        regression_path = Path.joinpath(dir_path, "regression.npy")
        if regression_path.exists():
            dmp = DynamicMovementPrimitives.load_dmp(dir_path)
            dmp.save_dmp(dir_path)
            return dmp

        dmp_data_path = Path.joinpath(dir_path, "dmp_data.json")
        with open(dmp_data_path, "r") as f:
            data = json.load(f)
        c_order = data["c_order"]
        alpha_z = np.array(data["alpha_z"])
        n_rfs = data["n_rfs"]
        goal_j = np.array(data["goal_j"])
        start_j = np.array(data["starting_j"])
        regression = np.array(data["regression"])
        return cls(
            regression,
            c_order,
            dmp_parameters=(alpha_z, n_rfs),
            goal_joints=goal_j,
            initial_joints=start_j,
        )

    @classmethod
    def load_dmp(cls, dir_path: Path) -> "DynamicMovementPrimitives":
        """
        Creates an DynamicMovementPrimitives with parameters loaded from a dmp param
        file and regression information loaded from a regression file.

        :param dir_path: path of the directory containing the data to load
        :return: DynamicMovementPrimitives with the loaded parameters
        """

        warnings.warn(
            "Loading data this way is deprecated and should be changed to `load`",
            DeprecationWarning,
        )

        # load regression data
        regression_path = Path.joinpath(dir_path, "regression.npy")
        if not regression_path.exists():
            raise RuntimeError("regression.npy does not exist!")
        reg = np.load(str(regression_path))
        # load dmp parameters
        parameters_path = Path.joinpath(dir_path, "dmp_parameters.json")
        if not parameters_path.exists():
            raise RuntimeError("dmp_parameters,json does not exist!")
        with open(parameters_path, "r") as f:
            data = json.load(f)
        c_order = data["c_order"]
        alpha_z = np.array(data["alpha_z"])
        n_rfs = data["n_rfs"]
        return cls(
            reg,
            c_order,
            dmp_parameters=(alpha_z, n_rfs),
        )

```

What do you think? The user gets notified he uses the wrong one but like (for now) we can use and convert the Ebikon dataset.

Reply:
Yes I totally agree! Nice example of backward compatibility

Reply:
Let me add it